src/Castle.Facilities.AspNetCore/WindsorRegistrationExtensions.cs

```diff
@@ -0,0 +1,184 @@
+﻿// Copyright 2004-2018 Castle Project - http://www.castleproject.org/
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Facilities.AspNetCore
+{
+	using System;
+	using System.Linq;
+
+	using Castle.Core;
+	using Castle.Facilities.AspNetCore.Resolvers;
+	using Castle.Facilities.AspNetCore.Activators;
+	using Castle.Facilities.AspNetCore.Contributors;
+	using Castle.MicroKernel.Lifestyle;
+	using Castle.MicroKernel.Registration;
+	using Castle.MicroKernel.Resolvers.SpecializedResolvers;
+	using Castle.MicroKernel;
+	using Castle.Windsor;
+
+	using Microsoft.AspNetCore.Builder;
+	using Microsoft.AspNetCore.Http;
+	using Microsoft.AspNetCore.Mvc;
+	using Microsoft.AspNetCore.Razor.TagHelpers;
+	using Microsoft.Extensions.DependencyInjection;
+
+	public static class WindsorRegistrationExtensions
+	{
+		/// <summary>
+		/// This exposes Windsor components to the IServiceCollection using the `CrossWired` registration extension. 
+		/// </summary>
+		/// <param name="services">ASP.NET Core service collection from Microsoft.Extensions.DependencyInjection</param>
+		/// <param name="container">Windsor container which IServiceCollection resolves from using factory methods</param>
+		public static void CrossWiresInto(this IWindsorContainer container, IServiceCollection services)
+		{
```

Comment:
@generik0 also thought we needed to throw a helpful exception here in case somebody used the [CrossWire](https://github.com/castleproject/Windsor/pull/394/files#diff-77a22f5542ffd29a9b211c43bfb976f3R83) extension without calling this.

Reply:
I agree.